czkawka_core/src/temporary.rs

```diff
@@ -176,88 +162,72 @@ impl Temporary {
 
                     for ed in &self.directories.excluded_directories {
                         if next_folder == *ed {
-                            is_excluded_dir = true;
-                            break;
+                            continue 'dir;
                         }
                     }
-                    if !is_excluded_dir {
-                        let mut found_expression: bool = false;
-                        for expression in &self.excluded_items.items {
-                            if Common::regex_check(expression, &next_folder) {
-                                found_expression = true;
-                                break;
-                            }
-                        }
-                        if found_expression {
-                            break;
+                    for expression in &self.excluded_items.items {
+                        if Common::regex_check(expression, &next_folder) {
+                            continue 'dir;
                         }
-                        folders_to_check.push(next_folder);
                     }
+                    folders_to_check.push(next_folder);
                 } else if metadata.is_file() {
                     let file_name_lowercase: String = match entry_data.file_name().into_string() {
                         Ok(t) => t,
                         Err(_) => continue,
                     }
                     .to_lowercase();
-                    let mut is_temporary_file: bool = false;
 
                     // Temporary files which needs to have dot in name(not sure if exists without dot)
                     let temporary_with_dot = ["#", "thumbs.db", ".bak", "~", ".tmp", ".temp", ".ds_store", ".crdownload", ".part", ".cache", ".dmp", ".download", ".partial"];
 
                     if file_name_lowercase.contains('.') {
                         for temp in temporary_with_dot.iter() {
                             if file_name_lowercase.ends_with(temp) {
-                                is_temporary_file = true;
+                                break;
                             }
                         }
+                        self.information.number_of_ignored_files += 1;
+                        continue 'dir;
                     }
```

Comment:
When file doesn't contains dot, then should be ignored
This code works fine

```
                    let mut found : bool = false;

                    if file_name_lowercase.contains('.') {
                        for temp in temporary_with_dot.iter() {
                            if file_name_lowercase.ends_with(temp) {
                                 found = true;
                                 break;
                            }
                        }
                    }
                    if !found {
                        self.information.number_of_ignored_files += 1;
                        continue 'dir;
                    }

```

Reply:
Rust prefers using `any()` over iterating and setting a flag. Would something like this work?
```rust
if !file_name_lowercase.contains('.') || !temporary_with_dot.iter().any(|f| file_name_lowercase.ends_with(f)) {
    self.information.number_of_ignored_files += 1;
    continue 'dir;
}

```

Reply:
Sure,
Tested it and works fine, and took only 4 lines instead 13 from my example.

Reply:
Thanks, fixed in latest commit.